spinoso-string/src/impls.rs

```diff
@@ -253,13 +253,13 @@ impl<I: SliceIndex<[u8]>> Index<I> for String {
 
     #[inline]
     fn index(&self, index: I) -> &Self::Output {
-        Index::index(&self.buf, index)
+        Index::index(self.inner.as_slice(), index)
```

Comment:
These haven't been implemented on the sub types, but they are simply relying on the slice. Worth delegating?

Reply:
this seems fine to me.